tools/mmp/driver.cs

```diff
@@ -1720,6 +1755,17 @@ static void ProcessNativeLibrary (HashSet<string> processed, string library, Lis
 			}
 		}
 
+		static void LipoLibrary (string name, string dest)
+		{
+			string archToRemove = arch == "i386" ? "x86_64" : "i386";
+
+			int ret = XcodeRun ("lipo", $"{dest} -remove {archToRemove} -output {dest}");
+			if (ret != 0)
+				throw new MonoMacException (5311, true, "lipo failed with an error code '{0}'. Check build log for details.", ret);
+			if (name != "MonoPosixHelper")
+				ErrorHelper.Warning (2108, $"{name} was stripped of architecture {archToRemove} to comply with App Store restrictions. This could break exisiting codesigning signatures. Consider stripping the library with lipo or disabling with --native-strip=skip");
```

Comment:
The "obvious" thing to do here is to first write a test that says "if I'm 64-bit and i have a 64-bit library there should be no warning".

And yes, it looks like that test would fail right now. :(